NetKet/Sampler/abstract_sampler.hpp

```diff
@@ -41,5 +41,19 @@ class AbstractSampler {
   virtual ~AbstractSampler() {}
 };
 
+template <class WfType>
```

Comment:
I think we can reasonably assume all samplers have a Seed method (at least I can't think now of cases where a generic sampler wouldn't have a seed method, but maybe you had something specific in mind?). 
I would suggest to avoid creating a `SeeableSampler` class and just move the `Seed` abstract method in the `AbstractSampler` class

Reply:
I wasn't completely sure whether every possible sampler needs random number generation, though I do not have a specific case in mind. So yes, I can move that to the base class.